soroban-sdk-macros/src/map_type.rs

```diff
@@ -37,6 +40,31 @@ pub fn map_type(t: &Type, allow_hash: bool) -> Result<ScSpecTypeDef, Error> {
                     "Address" => Ok(ScSpecTypeDef::Address),
                     "Timepoint" => Ok(ScSpecTypeDef::Timepoint),
                     "Duration" => Ok(ScSpecTypeDef::Duration),
+                    // The BLS types `G1Affine`, `G2Affine`, and `Fr` are represented in
+                    // the contract's interface by their underlying data types: I.e.
+                    // G1Affine/G2Affine => BytesN<N>, Fr => U256. This approach
+                    // simplifies integration with contract development tooling, as it
+                    // avoids introducing new spec types for these BLS constructs.
+                    //
+                    // While this is functionally sound because the BLS types are
+                    // essentially newtypes over their inner representations, it means
+                    // that the specific semantic meaning of `G1Affine`, `G2Affine`, or
+                    // `Fr` is not directly visible in the compiled WASM interface. For
+                    // example, a contract function expecting a `G1Affine` will appear
+                    // in the WASM interface as expecting a `BytesN<96>`.
+                    //
+                    // Future enhancements might allow the macro to automatically deduce
+                    // and utilize the inner types for types defined using the New Type
+                    // Idiom. For more details, see the tracking issue for supporting
+                    // type aliases:
+                    // https://github.com/stellar/rs-soroban-sdk/issues/1063
+                    "G1Affine" => Ok(ScSpecTypeDef::BytesN(ScSpecTypeBytesN {
+                        n: G1_SERIALIZED_SIZE,
+                    })),
+                    "G2Affine" => Ok(ScSpecTypeDef::BytesN(ScSpecTypeBytesN {
+                        n: G2_SERIALIZED_SIZE,
+                    })),
+                    "Fr" => Ok(ScSpecTypeDef::U256),
```

Comment:
@jayz22 Do the `Fp` and `Fp2` types need listing here too?

Reply:
Hmm good call. My initial intuition was no since these are just field elements that makes up the coordinates for G1 and G2. No need to store/pass them as function argument directly. 
But looking at it again, we do have host functions `bls12_381_map_fp2/1_to_g2/1` that maps a field element to a point. 
So I guess let's just add it for completeness.

Reply:
Added Fp and Fp2 and updated the test contract to invoke host functions using those.